### ISSUE TITLE
Form with array of objects automatically added with array of itself with Ids suffix

### ISSUE THREAD

**Issue by Tanapruk** [marmelab/react-admin#4067]
**What you were expecting:**

Clicking save after editing should send like.
```
{
 categories: [{
"id": "1234",
"name": "nice"
}]
}
```

**What happened instead:**
Clicking save after editing should send like.
a new field called categoriesIds are magically added and fail my graphql
```
{
 categories: [{
"id": "1234",
"name": "nice"
}],
categoriesIds: ["1234"]
}
```

**Steps to reproduce:**
It happened after upgrading to 3.0.0. My best guess is the react-final-form automatically created the params with Ids suffix, since it shows the ids inside <FormDataConsumer>. I try to create a new fields with array of objects but that do not show the problems.

For now, I fix it by removing fields with ids suffix at the dataProvider.

**Environment**

* React-admin version: 3.0.1
* Last version that did not exhibit the issue (if applicable): 2.98
* React version: 16.8
* Browser: Chrome
* Stack trace (in case of a JS error):

**Comment by djhi** [comment]
Thanks for reporting this. If you are able to illustrate the bug or feature request with an example, please provide a sample application via one of the following means:

* CodeSandbox (https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple)
* A sample application via GitHub

**Comment by wmwart** [comment]
@Tanapruk, This is normal work and React-admin feature.
Why does your GraphQL not work? I use GraphQL-Yoga for the backend and this solution not only does not break, but also helps in development.

However, there is one problem that I described [here](https://github.com/marmelab/react-admin/issues/4037). The React-admin also converts my JsonInput in the same way, which sends a custom data object to the Provider.
@djhi, what do you think about this?

**Comment by Tanapruk** [comment]
@wmwart I guess my problem is similar to yours but I couldn't make a minimum project to reproduce it right now.

**Comment by fzaninotto** [comment]
Should be fixed by #4103 4103
